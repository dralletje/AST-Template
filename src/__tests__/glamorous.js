let { template: t, match_inside, unparsable } = require('../ast-template.js');

console.log(`unparsable():`, unparsable`call(${['x']})`);

it('should match a static glamorous call', () => {
  let source = `
    let has_border = true;
    let component = glamorous.div({
      backgroundColor: 'black',
      color: 'red',
      border: has_border ? '2px solid black' : 'none',
    })
  `;

  let template = t.expression`
    glamorous.${t.Identifier('tag')}({
      ${t.many('entries',
        t.entry`
          ${t.Identifier('key')}: ${t.Expression('value')},
        `
      )}
    })
  `;

  expect(match_inside(source, template)).toMatchSnapshot();
})

it('should match a glamorous call with props', () => {
  let source = `
    let Card = glamorous.div({
      backgroundColor: 'black',
      color: 'red',
    }, (props) => {
      return {
        backgroundColor: props.backgroundColor,
      }
    });

    let Chart = glamorous.div({
      backgroundColor: 'black',
      color: 'red',
    });
  `;

  let template = t.expression`
    glamorous.${t.Identifier('tag')}(
      {
        ${t.many('entries',
          t.entry`
            ${t.Identifier('key')}: ${t.Expression('value')},
          `
        )}
      },
      ${t.optional('props-function',
        t.expression`(${t.Identifier('prop-arg')}) => {
          return {
            ${t.many('entries',
              t.entry`
                ${t.Identifier('key')}: ${t.Expression('value')},
              `
            )}
          }
        }`
      )}
    )
  `;

  expect(match_inside(source, t)).toMatchSnapshot();
})
