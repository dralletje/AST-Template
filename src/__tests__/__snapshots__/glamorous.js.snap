// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match a glamorous call with props 1`] = `Array []`;

exports[`should match a static glamorous call 1`] = `
Array [
  Object {
    "areas": Object {
      "entries": Array [
        Object {
          "key": "backgroundColor",
          "value": "\\"black\\"",
        },
        Object {
          "key": "color",
          "value": "\\"red\\"",
        },
        Object {
          "key": "border",
          "value": "has_border ? '2px solid black' : 'none'",
        },
      ],
      "tag": "div",
    },
  },
]
`;
